docs/aws/audit/sqsmonitoring/rules/queue_server_side_encryption_remediation.mdx

```diff
@@ -1,6 +1,210 @@
+   After creating or updating the queue, you can verify that SSE is enforced by retrieving the queue attributes.
+
+   ```python
+   # Retrieve the queue attributes to verify SSE configuration
+   response = sqs.get_queue_attributes(
+       QueueUrl=queue_url,
+       AttributeNames=['All']
+   )
+
+   print("Queue Attributes:")
+   for key, value in response['Attributes'].items():
+       print(f"{key}: {value}")
+   ```
+
+By following these steps, you can ensure that your SQS queues are configured to enforce Server Side Encryption using a specified KMS key. This helps in preventing the misconfiguration related to the lack of SSE in SQS queues.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the Amazon SQS console at https://console.aws.amazon.com/sqs/.
+2. In the navigation pane, choose 'Queues'.
+3. In the 'Queue Name' list, choose the name of the queue.
+4. In the 'Details' section, under 'Queue Attributes', check the 'Server-side encryption' attribute. If it is not set to 'Enabled', then the SQS queue is not enforcing server-side encryption.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   ```
+   pip install awscli
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all SQS queues: Use the following command to list all your SQS queues:
+
+   ```
+   aws sqs list-queues
+   ```
+   This command will return a list of URLs for each of your SQS queues.
+
+3. Get queue attributes: For each queue URL returned from the previous command, run the following command to get its attributes:
+
+   ```
+   aws sqs get-queue-attributes --queue-url <QueueURL> --attribute-names All
+   ```
+   Replace `<QueueURL>` with the URL of your queue. This command will return all the attributes of the specified queue.
+
+4. Check for Server Side Encryption: In the output of the previous command, look for the `KmsMasterKeyId` attribute. If this attribute is present and has a value, it means that Server Side Encryption is enabled for the queue. If this attribute is not present or has no value, it means that Server Side Encryption is not enabled for the queue.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>